### PR TITLE
fix(typescript): reexport Adapter* types from SvelteKit client pkg

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Since we want folks to only install the client pkg, they have no access to the `Adapter*` types they might want to use in their config setup, especially when writing their prisma/drizzle/whatever schema files.

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/frameworks-sveltekit/src/lib/index.ts [hunk 1]
```diff
@@ -215,6 +215,14 @@ export type {
...
 } from "@auth/core/types"
 
+export type {
+  Adapter,
+  AdapterAccount,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "@auth/core/adapters"
+
```
Comment (balazsorban44) [inline packages/frameworks-sveltekit/src/lib/index.ts:225]: To keep it consistent, maybe add `adapter.ts`?

Needs to be updated in package.json#exports too.

(you can verify via `pnpm pack`)

File: packages/frameworks-sveltekit/src/lib/index.ts [hunk 2]
```diff
@@ -266,7 +274,7 @@ const actions: AuthAction[] = [
  */
 export function SvelteKitAuth(
   config:
-    | SvelteKitAuthConfig
+    | (SvelteKitAuthConfig & { basePath?: string })
```
Comment (balazsorban44) [inline packages/frameworks-sveltekit/src/lib/index.ts:277]:
```suggestion
    | SvelteKitAuthConfig
```

Likely you forgot to run `pnpm dev` or `pnpm build` for `@auth/core`.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: packages/frameworks-sveltekit/src/lib/index.ts (hunk 1 above)
Comment (ndom91) [reply packages/frameworks-sveltekit/src/lib/index.ts:225]: Yeah like copying over the file liek with providers? I was thinking about it, but we dont need all the additional stuff, right - its really just these few types :thinking:

Comment (balazsorban44) [reply packages/frameworks-sveltekit/src/lib/index.ts:225]: No, create an `adapter.ts` file and put

```ts
export type {
  Adapter,
  AdapterAccount,
  AdapterSession,
  AdapterUser,
  VerificationToken,
} from "@auth/core/adapters"
```

in it.

File: packages/frameworks-express/src/index.ts [hunk 3]
```diff
@@ -151,6 +151,14 @@ export type {
...
 } from "@auth/core/types"
 
+export type {
+  Adapter,
+  AdapterAccount,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "@auth/core/adapters"
+
```
Comment (balazsorban44) [inline packages/frameworks-express/src/index.ts:161]: See https://github.com/nextauthjs/next-auth/pull/9713/files#r1463289981

Comment (ndom91) [reply packages/frameworks-express/src/index.ts]: https://github.com/nextauthjs/next-auth/pull/9732

File: packages/next-auth/src/index.tsx [hunk 4]
```diff
@@ -92,6 +92,14 @@ export type {
...
 } from "@auth/core/types"
 
+export type {
+  Adapter,
+  AdapterAccount,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "@auth/core/adapters"
+
```
[the remark this answers is not in the document]
Comment (ndom91) [reply packages/next-auth/src/index.tsx]: https://github.com/nextauthjs/next-auth/pull/9730

**Pull request merged**